### PR TITLE
SQL-842: Implement required Tableau date functions in connector dialect definition

### PR DESCRIPTION
**Pull request by varsha804**

### CODE REVIEW

**Review by nbagnard — APPROVED** [review]
Just a side comment, but you were able to test them out and make sure that things worked the way we expect them to, isn't it?

File: connector/dialect.tdd [hunk 1]
```diff
@@ -65,6 +65,91 @@
       <unagg-formula>(case when %1 IS NULL THEN 0 ELSE 1 END)</unagg-formula>
       <argument type='datetime' />
     </function>
+    <function group='date' name='DATEADD' return-type='datetime'>
```
Comment (bucaojit) [inline connector/dialect.tdd]: Let's switch the `function` to `date-function` and remove the `group='date'` for all these functions as is [done here](https://github.com/tableau/connector-plugin-sdk/blob/8d53af1b834aa1b3e5c38d57042b03aac5adc58a/samples/components/dialects/FullFunctionMap.tdd#L1535).  
Such as: `<date-function name='DATEADD' return-type='datetime'>`

File: connector/dialect.tdd [hunk 2]
```diff
@@ -65,6 +65,91 @@
...
+      <formula part='day'>DATETRUNC(DAY, %2, %3)</formula>
+      <formula part='hour'>DATETRUNC(HOUR, %2, %3)</formula>
+      <formula part='minute'>DATETRUNC(MINUTE, %2, %3)</formula>
+      <formula part='second'>DATETRUNC(SECOND, %2, %3)</formula>
+      <formula part='quarter'>DATETRUNC(QUARTER, %2, %3)</formula>
+      <formula part='week'>DATETRUNC(WEEK, %2, %3)</formula>
+      <argument type='localstr'/>
+      <argument type='datetime'/>
+      <argument type='localstr'/>
+    </function>
```
Comment (bucaojit) [inline connector/dialect.tdd]: We can move these added functions to the end of the '\<function-map\>' section as they will not have a `group`.

**Review by bucaojit — CHANGES_REQUESTED** [review]
Nice work getting the date functions completed! 

Just a few suggestions.

File: connector/dialect.tdd (hunk 1 above)
Comment (varsha804) [reply connector/dialect.tdd]: Done!

File: connector/dialect.tdd (hunk 2 above)
Comment (varsha804) [reply connector/dialect.tdd]: Done

**Review by bucaojit — APPROVED** [review]
LGTM! 💯

**Comment by varsha804** [comment]
evergreen merge

**Pull request merged**